Sming/SmingCore/Network/TcpServer.cpp

```diff
@@ -85,7 +85,7 @@ void TcpServer::setTimeOut(uint16_t waitTimeOut)
 bool TcpServer::listen(int port)
 {
 	if (tcp == NULL)
-		tcp = tcp_new();
+		initialize(tcp_new());
```

Comment:
I am missing something. Where is the tcp initialization when the TcpClient is used to connect?